### ISSUE TITLE
Possibly invalid dynamic invocations not reported

### ISSUE THREAD

**Issue by wouterj** [vimeo/psalm#5286]
https://psalm.dev/r/5eea61068e

Psalm does not seem to check dynamic invocations. The above code doesn't restrict `$key`, so the method may not exists at all. I think it makes sense if static analysis would catch these cases and I couldn't find such a request already for Psalm (but not sure about internal complexity required for this).

For reference, at least 2 of these invocations are catched by PHPstan: https://phpstan.org/r/ef21ff78-46ca-491c-8870-ed2d09235fa6

**Comment by muglug** [comment]
Yeah, Psalm currently doesn't warn you when calling an unknown methods.

This should probably be reported on level 1 with something like `VariableMethodCall` or similar.
